io-ballerina/file_string_io.bal

```diff
@@ -173,12 +173,66 @@ public function fileWriteJson(@untainted string path, json content) returns @tai
 # ```
 # + path - The path of the XML file
 # + content - XML content to write
+# + xmlOptions - XML writing options(XML entity type and DOCTYPE)
+# + fileWriteOption - file write option(`OVERWRITE` and `APPEND` are the possible values, and the default value is `OVERWRITE`)
 # + return - The null `()` value when the writing was successful or an `io:Error`
-public function fileWriteXml(@untainted string path, xml content) returns Error? {
-    var byteChannel = openWritableFile(path);
+public function fileWriteXml(@untainted string path, xml content, *XmlWriteOptions xmlOptions, FileWriteOption fileWriteOption = OVERWRITE) returns Error? {
+    WritableByteChannel|Error byteChannel;
+    xml writeContent = xml ``;
+    if (xmlOptions.xmlEntityType == DOCUMENT_ENTITY) {
+        if (fileWriteOption == APPEND) {
+            return error ConfigurationError("The APPEND operation not allowed with DOCUMENT");
+        }
+        if (xml:length(content) > 1) {
+            return error ConfigurationError("The DOCUMENT XML can only contains single root");
+        }
+        if (xmlOptions.doctype != ()) {
+            writeContent = xml:concat(populateDoctype(content, <XmlDoctype>xmlOptions.doctype), NEW_LINE, content);
+        } else {
+            writeContent = content;
+        }
+        byteChannel = openWritableFile(path);
+    } else {
+        if (fileWriteOption == APPEND) {
+            byteChannel = openWritableFile(path, APPEND);
+        } else {
+            byteChannel = openWritableFile(path);
+        }
+        writeContent = content;
+    }
+
     if (byteChannel is WritableByteChannel) {
-        return channelWriteXml(byteChannel, content);
+        return channelWriteXml(byteChannel, writeContent);
     } else {
         return byteChannel;
     }
 }
+
+function populateDoctype(xml content, XmlDoctype doctype) returns string {
+    // Generate <!DOCTYPE rootElementName PUBLIC|SYSTEM PublicIdentifier SystemIdentifier internalSubset>
+    string doctypeElement = "";
+    string startElement = "<!DOCTYPE ";
+    string endElement = ">";
```

Comment:
Shall we use a string template for this

Reply:
https://ballerina.io/learn/by-example/string-template.html

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/77/commits/c0d902a6e341c5c475b6834b65f5ab7eb5a82b07